### PR TITLE
Make the room directory join rooms by alias

### PR DESCRIPTION
**Pull request by dbkr**
Also makes some improvement on the multiple code path mess for joining rooms.

Untouched: joining rooms from the slash command, but this PR is probably already big enough.

### CODE REVIEW

**Comment by dbkr** [comment]
Fixes https://github.com/vector-im/vector-web/issues/1532

File: src/components/structures/MatrixChat.js [hunk 1]
```diff
@@ -424,40 +424,10 @@ module.exports = React.createClass({
                 }
                 break;
             case 'view_room_alias':
```
Comment (richvdh) [inline src/components/structures/MatrixChat.js]: view_room_alias now looks pretty similar to view_room. Is it worth merging them?

**Comment by richvdh** [comment]
ok, I think this is broadly sound, and certainly reduces complexity in the code, which can only be a good thing. This is certainly an improvement on what went before. Agreed that it's best to leave the slashcommand for a separate PR.

Thinking aloud: I still don't have a happy feeling about all these dances. I'd kinda like some documentation about which components have responsibility for dealing with IDs and aliases (eg: finding a room in the MatrixClient; joining a room if need be; picking the canonical alias which we show to a user) - and whether this changes at all with the various flows (/join; pick a room in the a directory; type a name in the directory; follow a link to a room; be invited; etc) * (peeking yes/no?) * (guest yes/no?) * (3pid magics yes/no?). That's only about 40 different combinations, should be easy :'(

Anyway, this lgtm.

**Comment by richvdh** [comment]
Actually: you might want to merge `view_room_alias` and `view_room`, and consider my comment https://github.com/vector-im/vector-web/pull/1615#discussion_r66586171 before merging this

**Comment by dbkr** [comment]
ptal

**Comment by richvdh** [comment]
lgtm

**Pull request merged**